Mathlib/Data/Nat/PSub.lean

```diff
@@ -0,0 +1,122 @@
+/-
+Copyright (c) 2017 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro
+-/
+import Mathlib.Data.Option.Basic
+import Mathlib.Data.Nat.Basic
+
+/-!
+# Partial predecessor and partial subtraction on the natural numbers
+
+The usual definition of natural number subtraction (`nat.sub`) returns 0 as a "garbage value" for
+`a - b` when `a < b`. Similarly, `nat.pred 0` is defined to be `0`. The functions in this file
+wrap the result in an `option` type instead:
+
+## Main definitions
+
+- `nat.ppred`: a partial predecessor operation
+- `nat.psub`: a partial subtraction operation
+
+-/
+
+
+namespace Nat
+
+/-- Partial predecessor operation. Returns `ppred n = some m`
+  if `n = m + 1`, otherwise `none`. -/
+@[simp]
```

Comment:
Please remove `@[simp]` here and on `psub`. You may need to write the simp lemmas for the branches yourself.

Putting `@[simp]` on a definition by matching has changed behaviour in Lean 4.